csrc/multidevice/device_mesh.cpp

```diff
@@ -127,28 +120,47 @@ DeviceIdxType DeviceMesh::maxDeviceId() const {
 }
 
 namespace {
+// Maps a parallel type to axis. Returns -1 if the parallel type is
+// not in the device mesh.
 int64_t ptypeToAxis(ParallelType ptype, int64_t ndims) {
```

Comment:
Since you are at this, do you mind making ptypeToAxis a method of DeviceMesh? This way, users don't have to remember to pass in `rank()`.